DatadogCore/Tests/Datadog/TracerTests.swift

```diff
@@ -725,327 +725,88 @@ class TracerTests: XCTestCase {
 
     // MARK: - Injecting span context into carrier
 
-    func testItInjectsSpanContextWithHTTPHeadersWriter() {
-        Trace.enable(with: config, in: core)
-        let tracer = Tracer.shared(in: core)
-        let spanContext1 = DDSpanContext(traceID: .init(idHi: 10, idLo: 100), spanID: 200, parentSpanID: .mockAny(), baggageItems: .mockAny(), sampleRate: .mockRandom(), isKept: .mockRandom())
```

Comment:
🧹 I'm removing these tests from integration layer. Headers encoding / decoding was moved to `DatadogTrace` unit tests. This includes adding missing UT coverage for `HTTPHeadersWriter`.